### ISSUE TITLE
Can't update Concrete5

### ISSUE THREAD

**Issue by faminecode** [turnkeylinux/tracker#711]
Running a fresh Concrete5 14.1:

Getting "The directory /var/www/concrete5/updates must be writable by the web server." when attempting Concrete5 internal update mechanism.  Tried chown to www-data on the updates folder, but this results in an HTTP500 when attempting the download.

**Comment by JedMeister** [comment]
Hi, thanks for reporting. I'll have to have a look into this ASAP. Please note though that we are currently doing billing migrations for legacy Hub account holders so have a higher than usual support load. It may be another week or 2 (or perhaps longer) before I get a chance to look into this. I'll update this thread as soon as I have anything to add. But please feel free to bump this thread if you'd like an update.

**Comment by JedMeister** [comment]
Really sorry that I never got back to you on this. I've been swamped and only just came across it now while cleaning up the tracker. Pretty poor really...

Anyway, I'm almost certain that it could be worked around by giving the webserver ownership of the files. Something like:
````
chown -R www-data:www-data /var/www/concrete5
````
Then if you want to retighten it afterwards, then do this:
````
WEBROOT=/var/www/concrete5
chown -R root:root $WEBROOT
chown -R www-data:www-data $WEBROOT/application/files
chown -R www-data:www-data $WEBROOT/application/config
chown -R www-data:www-data $WEBROOT/packages
````
We've already released the new v14.2 appliance, so I've retagged this issue to v15.0. TBH though, I think that the right way to resolve this (and similar issues on other appliances) is a confconsole plugin to relax and tighten permissions (essentially what I noted above). Then you can have the security advantage of how we normally ship it; and relatively easily take advantage of the convenience of web UI updates.

**Comment by OnGle** [comment]
Just a little note to add to this, the directory that requires write permissions by www-data is `/var/www/concrete5/updates`. You may also need to increase your php memory limit in `/etc/php5/apache2/php.ini` to 128M.
